app/controllers/agents_controller.rb

```diff
@@ -237,14 +240,14 @@ def redirect_back(message, options = {})
       if @agent && !@agent.destroyed?
         path = agent_path(@agent)
```

Comment:
Could you please add `else path = agents_path` here?  It should fix the 404 bug.